doyensec/detectors/rce/torchserve/src/test/java/com/google/tsunami/plugins/detectors/rce/torchserve/TorchServeExploiterTest.java

```diff
@@ -0,0 +1,195 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.rce.torchserve;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.tsunami.common.data.NetworkEndpointUtils.forHostnameAndPort;
+
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.Software;
+import com.google.tsunami.proto.TransportProtocol;
+import java.io.IOException;
+import javax.inject.Inject;
+import okhttp3.mockwebserver.MockResponse;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TorchServeExploiterTest extends TorchServeManagementApiTestBase {
+  @Inject private TorchServeExploiter exploiter;
+
+  NetworkService service;
+
+  @Before
+  public void setUpNetworkService() {
+    service =
+        NetworkService.newBuilder()
+            .setNetworkEndpoint(
+                forHostnameAndPort(mockTorchServe.getHostName(), mockTorchServe.getPort()))
+            .setTransportProtocol(TransportProtocol.TCP)
+            .setSoftware(Software.newBuilder().setName("torchserve"))
+            .setServiceName("http")
+            .build();
+  }
+
+  private void enqueueMockTorchServeResponse(String response) {
+    mockTorchServe.enqueue(new MockResponse().setResponseCode(200).setBody(response));
+  }
+
+  private String API_DESCRIPTION_RESPONSE =
+      "{\n"
+          + "  \"openapi\": \"3.0.1\",\n"
+          + "  \"info\": {\n"
+          + "    \"title\": \"TorchServe APIs\",\n"
+          + "    \"description\": \"TorchServe is a flexible and easy to use tool for serving deep"
+          + " learning models\",\n"
+          + "    \"version\": \"0.8.1\"\n"
+          + "  },\n"
+          + "  \"paths\": {\n"
+          + "    \"/models\": {\n"
+          + "      \"post\": {\n"
+          + "        \"description\": \"Register a new model in TorchServe.\",\n"
+          + "        \"operationId\": \"registerModel\"\n"
+          + "     }\n"
+          + "    }\n"
+          + "  }\n"
+          + "}";
+
+  private String EMPTY_MODELS_RESPONSE = "{\"models\": []}";
+
+  private String getCustomizedMetadataResponse(@Nullable String metadata) {
+    return "[{ \"customizedMetadata\" : \"" + (metadata == null ? "" : metadata) + "\" }]";
+  }
+
+  @Test
+  public void isServiceVulnerable_returnsNullIfServiceIsNotTorchServe() throws IOException {
```

Comment:
Please consider rewriting the tests in the format `fuction_condition_result`, for example here: `isServiceVulnerable_whenBasicAndVulnerable_returnsNull`